conjure-runtime/src/builder.rs

```diff
@@ -433,3 +430,23 @@ pub enum Idempotency {
     /// No requests are assumed to be idempotent.
     Never,
 }
+
+/// Specifies the strategy used to select a node of a service to use for a request attempt.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum NodeSelectionStrategy {
+    /// Pin to a single host as long as it continues to successfully respond to requests.
+    ///
+    /// If the pinned node fails to successfully respond, the client will rotate through the other nodes until it finds
+    /// one that can successfully respond and then pin to that new node. The pinned node will also be randomly rotated
+    /// periodically to help spread load across the cluster.
+    ///
+    /// This is the default behavior.
+    PinUntilError,
+
+    /// Like `PinUntilError` except that the pinned node is never randomly shuffled.
+    PinUntilErrorWithoutReshuffle,
+
+    /// For each new request, select the "next" node (in some unspecified order).
+    Balanced,
```

Comment:
This is called "ROUND_ROBIN" in the c-j-r service config API, but Dialogue uses the term Balanced, including in the server strategy hinting response header so I chose that here.